src/paras/Shop.java

```diff
@@ -0,0 +1,109 @@
+package paras;
+
+public class Shop {
+	int maggieQuantity; 
+	int dosaQuantity; 
+	int pouchesQuantity; 
+	int panipuriQuantity; 
+	int masalaQuantity;
+	
+	void setInitialQ(int maggieQ, int dosaQ, int pouchesQ, int panipuriQ, int masalaQ){
+		
+		maggieQuantity = maggieQ;
+		dosaQuantity = dosaQ;
+		pouchesQuantity = pouchesQ;
+		panipuriQuantity = panipuriQ;
+		masalaQuantity = masalaQ;
+	}
+	
+	void requestedQuantity(int maggieQ, int dosaQ, int pouchesQ, int panipuriQ, int masalaQ, boolean partialAllowed){
+		
+		if(maggieQuantity < 1){
+			System.out.println("\n Maggie is Out of Stock");
+		}else if(maggieQuantity >= maggieQ){
+			System.out.println("\n As per your request, We will serve Maggie with " + maggieQ + " quantity");
+			maggieQuantity = maggieQuantity - maggieQ;
+		}else if(partialAllowed){
+			System.out.println("\n We served partial Maggie Quantity and count is " + maggieQ);
+			maggieQuantity = 0;
```

Comment:
If request stock is not available mean we are setting direct it zero, 
what if second time your calling same method and let say you request 20 packs, then this looks not logically correct